httpserver/onion_processing.h

```diff
@@ -0,0 +1,69 @@
+#pragma once
+
+#include <variant>
+
+namespace oxen {
+
+struct CiphertextPlusJson {
+    std::string ciphertext;
+    std::string json;
+};
+
+/// The request is to be forwarded to another SS node
+struct RelayToNodeInfo {
+    /// Inner ciphertext for next node
+    std::string ciphertext;
+    // Key to be forwarded to next node for decryption
+    std::string ephemeral_key;
+    // Next node's ed25519 key
+    std::string next_node;
+};
+
+std::ostream& operator<<(std::ostream& os, const RelayToNodeInfo& p);
+
+bool operator==(const RelayToNodeInfo& lhs, const RelayToNodeInfo& rhs);
+
+/// The request is to be forwarded to some non-SS server
+/// that supports our protocol (e.g. Session File Server)
+struct RelayToServerInfo {
+    // Result of decryption (intact)
+    std::string payload;
+    // Server's address
+    std::string host;
+    // Server's port
+    uint16_t port;
+    // Http or Https (TODO: use enum)
+    std::string protocol;
+    // Request's target
+    std::string target;
+};
+
+std::ostream& operator<<(std::ostream& os, const RelayToServerInfo& p);
+
+bool operator==(const RelayToServerInfo& lhs, const RelayToServerInfo& rhs);
+
+/// We are the final destination for this request
+struct FinalDestinationInfo {
+    std::string body;
+};
+
+std::ostream& operator<<(std::ostream& os, const FinalDestinationInfo& p);
+
+bool operator==(const FinalDestinationInfo& lhs,
+                const FinalDestinationInfo& rhs);
+
+enum class ProcessCiphertextError {
+    INVALID_CIPHERTEXT,
+    INVALID_JSON,
+};
+
+using ParsedInfo = std::variant<RelayToNodeInfo, RelayToServerInfo,
+                                FinalDestinationInfo, ProcessCiphertextError>;
+
+// TODO: move this from http_connection.h after refactoring
+auto parse_combined_payload(const std::string& payload) -> CiphertextPlusJson;
+
+auto process_inner_request(const CiphertextPlusJson& parsed,
+                           std::string plaintext) -> ParsedInfo;
```

Comment:
These trailing return types make these functions inconsistent with the style of everything else in the file, and feel a bit like attempting to abuse a C++ feature (meant for a deduced argument-dependent return type) to make code feel more like Rust.

Reply:
I thought the C++ community was advocating for using trailing returns universally, whether you strictly need it or not. Where is the abuse? Sure, most of the rest of the code is not using this feature yet, but the same goes for other new features like `std::string_vew` which get added incrementally. Personally, I like trailing returns because they the names of the functions nicely, which is useful for when the return type is long. To me it outweights the benefits of consistency.